### DISCUSSION TITLE
How to import v0.22.x schema to v0.23 ?

### DISCUSSION THREAD

**Discussion by engageintellect** [pocketbase/pocketbase#6017] [Q&A] [answered]
I am trying to update my app to v0.23. Every time I try to import my schema, I get no errors, and all of my collections are there, but their props (columns ?) are not... What am I missing here?

I am using my own self-hosted instance of PB, along with the JS client for my app, any help/tips are much appreciated.

**Comment by ganigeorgiev** [comment]
v0.23 comes with many breaking changes and you can't import a v0.22.x schema without modifications.

Please follow the v0.23 upgrade notes from the [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v0230) and its linked [Go](https://pocketbase.io/v023upgrade/go/) or [JSVM](https://pocketbase.io/v023upgrade/jsvm/) guides.

If you just want to upgrade the schema, then you can try to:
1. Comment all custom hooks and migrations code (e.g. temp rename the folders)
2. Run the v0.23.x executable against your v0.22.x `pb_data` (it comes with system migration that will adjust the old db structure to the new one).
3. Login in the dashboard and generate a new collections export (it should contain the modification from step 2).

**Reply by engageintellect** [reply]
Thank you for your reply, you absolutely helped point me in the right direction. I'd really appreciate if you could help me with this last piece...

I am using Pocketbase in my Sveltekit app.

So, I ended up just installing v0.23 locally, and replicating my schema by hand. Then, I upgraded the pocketbase client in my Sveltekit app to v0.22.

Everything appears to be working - with no changes to any of my existing logic. However, I am getting one deprecation error on a piece of code that I have isolated:

<img width="944" alt="Screenshot 2024-12-04 at 9 24 11 PM" src="https://github.com/user-attachments/assets/7e4cf2ff-345d-4d3c-b4d2-df9053b435b4">


console logging "locals.pb.authStore" returns everything I am expecting inside of a "LocalAuthStore" object. However, when I try to console log "locals.pb.authStore.LocalAuthStore", I get this error:

<img width="903" alt="Screenshot 2024-12-05 at 6 32 14 PM" src="https://github.com/user-attachments/assets/f76a3d68-fa88-4af1-b491-3b41e4e77c3a">


After searching the docs, I wasn't able to find what this deprecation has been replaced with, but I can't for the life of me figuring out the shape of this.

EDIT: After a little more digging, I see that "baseModel", which is what I'm guessing I would use in place of "model" is a protected class. I can get around this by overriding TypeScript checks and using an indexer, like this:

```typescript
console.log('user id:', locals.pb.authStore['baseModel']?.id);
```
 However, I am sure that is a better/cleaner way I should be accessing this stuff.

**Reply by ganigeorgiev** [reply, marked as answer]
Please read the linked CHANGELOG (and more specifically https://github.com/pocketbase/js-sdk/blob/master/CHANGELOG.md).

**Reply by engageintellect** [reply]
> Please read the linked CHANGELOG (and more specifically https://github.com/pocketbase/js-sdk/blob/master/CHANGELOG.md).

Got everything running perfectly! Thank you for your help, sorry I overlooked the docs. 😅
